framework/src/utils/BicubicSplineInterpolation.C

```diff
@@ -0,0 +1,226 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "BicubicSplineInterpolation.h"
+#include "MooseError.h"
+
+int BicubicSplineInterpolation::_file_number = 0;
+
+BicubicSplineInterpolation::BicubicSplineInterpolation()
+{
+}
+
+BicubicSplineInterpolation::BicubicSplineInterpolation(const std::vector<Real> & x1, const std::vector<Real> & x2, const std::vector<std::vector<Real> > & y, const std::vector<Real> & yx11, const std::vector<Real> & yx1n, const std::vector<Real> & yx21, const std::vector<Real> & yx2n) :
+    SplineInterpolationBase(),
+    _x1(x1),
+    _x2(x2),
+    _y(y),
+    _yx11(yx11),
+    _yx1n(yx1n),
+    _yx21(yx21),
+    _yx2n(yx2n)
+{
+  errorCheck();
+  solve();
+}
+
+void
+BicubicSplineInterpolation::setData(const std::vector<Real> & x1, const std::vector<Real> & x2, const std::vector<std::vector<Real> > & y, const std::vector<Real> & yx11, const std::vector<Real> & yx1n, const std::vector<Real> & yx21, const std::vector<Real> & yx2n)
+{
+  _x1 = x1;
+  _x2 = x2;
+  _y = y;
+  _yx11 = yx11;
+  _yx1n = yx1n;
+  _yx21 = yx21;
+  _yx2n = yx2n;
+  errorCheck();
+  solve();
+}
+
+void
+BicubicSplineInterpolation::errorCheck()
+{
+  unsigned int m = _x1.size(), n = _x2.size();
+
+  if (_y.size() != m)
+    mooseError("y row dimension does not match the size of x1.");
+  else
+    for (int i = 0; i < _y.size(); ++i)
+      if (_y[i].size() != n)
+        mooseError("y column dimension does not match the size of x2.");
+
+  if (_yx11.empty())
+    _yx11.resize(n, 1e30);
```

Comment:
Yea, it seemed like the most natural place to put it though and I didn't want to change the method name. If you have a preferred way to do it though, I can definitely do it that way